run_action.py

```diff
@@ -495,6 +494,138 @@ def dismiss_change_requests(
         # Avoid triggering abuse detection
         time.sleep(10)
 
+    if auto_resolve_conversations:
+        resolve_conversations(
+            github_token=github_token,
+            repo=repo,
+            pull_request_id=pull_request_id,
+            github_api_timeout=github_api_timeout,
+        )
+
+
+def conversation_threads_to_close(repo, pr_number, github_token, github_api_timeout):
+    """Generator of unresolved conversation threads to close
+
+    Uses the GitHub GraphQL API to get conversation threads for the given PR.
+    Then filters for unresolved threads and those that have been created by the action.
+    """
+
+    repo_owner, repo_name = repo.split("/")
+    query = """
+    query {
+      repository(owner: "%s", name: "%s") {
+        pullRequest(number: %d) {
+          id
+          reviewThreads(last: 100) {
+            nodes {
+              id
+              isResolved
+              comments(first: 1) {
+                nodes {
+                  id
+                  body
+                  author {
+                    login
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """ % (
+        repo_owner,
+        repo_name,
+        pr_number,
+    )
```

Comment:
Nitpick: Can you please use an `F string`  for clarity since this is rather long query?

Reply:
Done. With the other query too. Had opted to use this format to not need to use double curlies.

Reply:
Oh jeez, looks ugly now. Is it OK to take my suggestion back? :laughing:

Reply:
Sure. I've made the change a sep commit. I'll just drop that and force-push :)

Reply:
Done